src/api/app/jobs/update_released_binaries_job.rb

```diff
@@ -19,7 +19,120 @@ def update_binary_releases(repository, key, time = Time.now)
+          entry.modify_time = time
+          entry.save!
+          hash[:operation] = 'modified' # new entry will get "modified" instead of "added"
+        end
+
+        # complete hash for new entry
+        hash[:binary_releasetime] = time
+        hash[:binary_id] = binary['binaryid'] if binary['binaryid'].present?
+        hash[:binary_buildtime] = nil
+        hash[:binary_buildtime] = Time.strptime(binary['buildtime'].to_s, '%s') if binary['buildtime'].present?
+        hash[:binary_disturl] = binary['disturl']
+        hash[:binary_supportstatus] = binary['supportstatus']
+        hash[:binary_cpeid] = binary['cpeid']
+        if binary['updateinfoid']
+          hash[:binary_updateinfo] = binary['updateinfoid']
+          hash[:binary_updateinfo_version] = binary['updateinfoversion']
+        end
+        if binary['project'].present? && binary['package'].present?
+          # the package may be missing if the binary comes via DoD
+          source_package = Package.striping_multibuild_suffix(binary['package'])
+          rp = Package.find_by_project_and_name(binary['project'], source_package)
+          if source_package.include?(':') && !source_package.start_with?('_product:')
+            flavor_name = binary['package'].gsub(/^#{source_package}:/, '')
+            hash[:flavor] = flavor_name
+          end
+          hash[:release_package_id] = rp.id if binary['project'] && rp
+        end
+        if binary['patchinforef']
+          begin
+            patchinfo = Patchinfo.new(data: Backend::Api::Sources::Project.patchinfo(binary['patchinforef']))
+          rescue Backend::NotFoundError
+            # patchinfo disappeared meanwhile
+          end
+          hash[:binary_maintainer] = patchinfo.hashed['packager'] if patchinfo && patchinfo.hashed['packager']
+        end
+
+        # put a reference to the medium aka container
+        hash[:on_medium] = medium_hash[binary['medium']] if binary['medium'].present?
+
+        # new entry, also for modified binaries.
+        entry = repository.binary_releases.create(hash)
+        processed_item[entry.id] = true
+
+        # store in medium case
+        medium_hash[binary['ismedium']] = entry if binary['ismedium'].present?
+      end
+
+      # and mark all not processed binaries as removed
+      BinaryRelease.where(repository: repository, obsolete_time: nil, modify_time: nil).where.not(id: processed_item.keys).update_all(obsolete_time: time)
+    end
+  end
+
+  def hashkey_db(binary)
+    "#{binary['binary_name']}|#{binary['binary_version'] || '0'}|#{binary['binary_release'] || '0'}|#{binary['binary_epoch'] || '0'}|#{binary['binary_arch'] || ''}|#{binary['medium'] || ''}"
+  end
+
+  def hashkey_json(binary, medium)
+    "#{binary['name']}|#{binary['version'] || '0'}|#{binary['release'] || '0'}|#{binary['epoch'] || '0'}|#{binary['binaryarch'] || ''}|#{medium || ''}"
+  end
+
+  def old_and_new_binary_identical?(old_binary, new_binary)
```

Comment:
This changed it's name and now get's bassed an activerecord and a hash (before it was an instance method of `BinaryRelease`)